### PR TITLE
Update version for the next release (v0.9.0)

### PR DESCRIPTION
**Pull request by alallema**
## 💥  Breaking Changes

- `SearchQuery` changes (#505) @alallema
  - Renamed `facetsDistribution` field to `facets`
  - Renamed `matches` field to `showMatchesPosition`
- `SearchResult` changes (#505) @alallema
  - Renamed `nbHits ` field to `estimatedTotalHits`
  - Removed `exhaustiveFacetsCount` field
  - Removed `exhaustiveNbHits` field
- Renamed `Result` class to plural `Results` (#508) @alallema 
- Remove method `client.getDumpStatus` (#515) @alallema
- Method `client.createDump` now returns a `TaskInfo` object (#515) @alallema
- `client.getTasks()` now returns a `TasksResults` (#517) @alallema
- `client.getIndexes()` now returns a `Results<Index>` (#516) @alallema 
- `client.getRawIndexes()` method return a String (#516) @alallema 
- `client.getKeys()` now returns a `Results<Key>` (#517) @alallema
- `client.getDocuments()` now the method returns a `Results<T>` (#518) @alallema
- Create two separate methods for `getDocument`:
    - `<T> Results<T> getDocument(String identifier, Class<T> targetClass)
    - `String getRawDocument(String identifier)`
- Create two separate methods for `getDocument`:
    - `<T> Results<T> getDocuments(Class<T> targetClass)`
    - `String getRawDocuments()`
- Remove `client.getRawIndex()` method (#516) @alallema 
- Add new `:name` and `:uid` attributes in `Key` class (#517) @alallema 
- `client.generateTenantToken` method now waits for an `apiKeyUid` (Uid of a signing API key) as parameter.

## 🚀 Enhancements

* Developer experience enhancements (#502) @brunoocasali
* Creation of `KeyUpdate` class (#517) @alallema 
* `client.getTasks()` now accepts an object with pagination `TasksQuery` (#510) @alallema 
* `client.getIndexes()` now accepts an object with pagination `IndexesQuery` (#516) @alallema 
* `client.getKeys()` now accepts an object with pagination `KeysQuery` (#517) @alallema 
- `client.getDocuments()`, `client.getDocument()` Add the possibility to reduce the body payload by now accepting a query parameter called `fields` (previously called `attributesToRetrieve`) (#518) @alallema
* Add User-Agent header to have analytics in every http request (#503) @brunoocasali
Analytics is enabled by default in the server, but you can disable them by following [this guide](https://docs.meilisearch.com/learn/what_is_meilisearch/telemetry.html#how-to-disable-data-collection)
Also, of course, every analytics data we collect are **ANONYMOUS** [read the guide for more information](https://docs.meilisearch.com/learn/what_is_meilisearch/telemetry.html).

Thanks again to @alallema, @brunoocasali ! 🎉

### CODE REVIEW

**Comment by brunoocasali** [comment]
@alallema we need to add a disclaimer about the "analytics":

You can copy from here: https://github.com/meilisearch/meilisearch-dart/releases/tag/v0.5.1

**Comment by brunoocasali** [comment]
Can you convert these two:

```
Remove GET /indexes/:indexUid/tasks. Use GET /tasks?indexUid=:indexUid instead (https://github.com/meilisearch/meilisearch-java/pull/510) @alallema
Remove GET /indexes/:indexUid/tasks/:taskUid. Use GET /tasks/:taskUid instead (https://github.com/meilisearch/meilisearch-java/pull/510) @alallema
```

To use the Java methods instead of the HTTP routes?

**Comment by alallema** [comment]
> Can you convert these two:
> 
> ```
> Remove GET /indexes/:indexUid/tasks. Use GET /tasks?indexUid=:indexUid instead (https://github.com/meilisearch/meilisearch-java/pull/510) @alallema
> Remove GET /indexes/:indexUid/tasks/:taskUid. Use GET /tasks/:taskUid instead (https://github.com/meilisearch/meilisearch-java/pull/510) @alallema
> ```
> 
> To use the Java methods instead of the HTTP routes?

It's because the method didn't really changes is the route inside the method that has changed no?

**Comment by brunoocasali** [comment]
There is just a small typo in the backticks here: 
>>> `GET /documents/:uid`` now the return is wrap in a result 

And also, you can change it to `now the method returns a Results<T>` and of course, change the `GET /documents/:uid` to the Java method :)

**Comment by brunoocasali** [comment]
Here `Create two separate methods for` you can omit the `-> return an...`  because the users will understand it just by looking to the method signature you use:

Here (blue part): image

It will be like this:
image

--- 
And here `In the endpoint GET /tasks we can filter by` you can change to use the Java method as well :)

**Comment by brunoocasali** [comment]
> It's because the methods didn't really changes is the route inside the method that has changed no?

@alallema if nothing related to the method signature changed you can remove the entire change from the changelog, even if Meili stopped to accept that method, we manage to keep the same behavior, so no need to inform the user about that.

File: README.md [hunk 1]
```diff
@@ -56,7 +56,7 @@ Add the following code to the `<dependencies>` section of your project:
 <dependency>
   <groupId>com.meilisearch.sdk</groupId>
   <artifactId>meilisearch-java</artifactId>
-  <version>0.8.0</version>
```
Comment (brunoocasali) [inline README.md:59]: It is missing one more place:

`[.code-samples.meilisearch.yaml](https://github.com/meilisearch/meilisearch-java/blob/main/.code-samples.meilisearch.yaml) in the getting_started_add_documents_md field`

**Review by brunoocasali — APPROVED** [review]
LGTM! 🔥🔥

**Pull request merged**